### PR TITLE
MakeRValue should be applied to tuple literals recursively

### PR DESCRIPTION
**Pull request by VSadov**
Fixes:#15198

### CODE REVIEW

**Comment by VSadov** [comment]
@dotnet/roslyn-compiler  - please review (bug blocks some use of tuples in Roslyn.sln)

File: src/Compilers/VisualBasic/Portable/Binding/Binder_Expressions.vb [hunk 1]
```diff
@@ -1242,6 +1242,28 @@ Namespace Microsoft.CodeAnalysis.VisualBasic
...
                 Return memberAccess.Update(enclosed)
 
+            ElseIf expr.Kind = BoundKind.TupleLiteral AndAlso expr.Type IsNot Nothing Then
+                Dim tupleLiteral = DirectCast(expr, BoundTupleLiteral)
+                Dim args = tupleLiteral.Arguments
+                Dim rvalues As ArrayBuilder(Of BoundExpression) = Nothing
+
+                For i As Integer = 0 To args.Length - 1
+                    Dim arg = args(i)
+                    Dim rvalue = MakeRValue(arg, diagnostics)
```
Comment (AlekseyTs) [inline src/Compilers/VisualBasic/Portable/Binding/Binder_Expressions.vb]: >Dim rvalue = MakeRValue(arg, diagnostics) [](start = 20, length = 41)

It feels to me that we shouldn't need to do this for properties here, it should be done when we are binding elements.

File: src/Compilers/VisualBasic/Test/Emit/CodeGen/CodeGenTuples.vb [hunk 2]
```diff
@@ -17907,6 +17907,66 @@ BC42024: Unused local variable: 'x6'.
 
         End Sub
 
+        <Fact>
+        <WorkItem(15198, "https://github.com/dotnet/roslyn/issues/15198")>
+        Public Sub TuplePropertyArgs001()
```
Comment (AlekseyTs) [inline src/Compilers/VisualBasic/Test/Emit/CodeGen/CodeGenTuples.vb]: >TuplePropertyArgs001 [](start = 19, length = 20)

All tests here are for properties. Is this the only possible source of problems?

Comment (VSadov) [inline src/Compilers/VisualBasic/Test/Emit/CodeGen/CodeGenTuples.vb]: I think so. I have also added a test for latebound properties.

File: src/Compilers/VisualBasic/Portable/Binding/Binder_Expressions.vb [hunk 3]
```diff
@@ -354,13 +354,17 @@ Namespace Microsoft.CodeAnalysis.VisualBasic
...
-                boundArguments.Add(boundArgument)
-
                 Dim elementType = GetTupleFieldType(boundArgument, argumentSyntax, diagnostics, hasNaturalType, hasErrors)
+
                 If elementType Is Nothing Then
                     hasInferredType = False
                 End If
 
+                If boundArgument.Type IsNot Nothing Then
+                    boundArgument = MakeRValue(boundArgument, diagnostics)
```
Comment (AlekseyTs) [inline src/Compilers/VisualBasic/Portable/Binding/Binder_Expressions.vb]: >boundArgument = MakeRValue(boundArgument, diagnostics) [](start = 20, length = 54)

Consider doing this before calling GetTupleFieldType. This probably won't change anything in terms of behavior though.

**Comment by AlekseyTs** [comment]
LGTM

File: src/Compilers/VisualBasic/Portable/Binding/Binder_Expressions.vb (hunk 3 above)
Comment (VSadov) [inline src/Compilers/VisualBasic/Portable/Binding/Binder_Expressions.vb]: GetTupleFieldType does not change boundArgument (it is passed byval and is immutable)
I think forcing it into RValue before storing into builder is mildly better.

**Comment by CyrusNajmabadi** [comment]
Shoudl this be in RC.2?

**Pull request merged**